src/Interpreters/InterpreterInsertQuery.cpp

```diff
@@ -735,26 +733,33 @@ QueryPipeline InterpreterInsertQuery::buildInsertSelectPipeline(ASTInsertQuery &
 
         new_settings[Setting::max_threads] = std::max<UInt64>(1, settings[Setting::max_insert_threads]);
```

Comment:
In the test below, we set `max_insert_threads=4` and `max_threads=8`, but according to this line `max_threads` should be overwritten, so for insert select pipeline,`max_threads` value should be `4` but it's `8`. Do I miss something or?